lib/MongoDB/Collection.pm

```diff
@@ -1000,6 +1001,79 @@ sub find_one_and_update {
     return $self->_find_one_and_update_or_replace($filter, $update, $options);
 }
 
+=method watch
+
+Watches for changes on this collection-
+
+Perform an aggregation with an implicit initial C<$changeStream> stage
+and returns a L<MongoDB::ChangeStream> result which can be used to
+iterate over the changes in the collection. This functionality is
+available since MongoDB 3.6.
+
+    my $stream = $collection->watch();
+    my $stream = $collection->watch( \@pipeline );
+    my $stream = $collection->watch( \@pipeline, \%options );
+
+    while (my $change = $stream->next) {
+        # process $change
+    }
+
+The returned stream will not block forever waiting for changes. If you
+want to respond to changes over a longer time use C<maxAwaitTimeMS> and
+regularly call C<next> in a loop.
+
+B<Note>: Using this helper method is preferred to manually aggregating
+with a C<$changeStream> stage, since it will automatically resume when
+the connection was terminated.
+
+The optional first argument must be an array-ref of
+L<aggregation pipeline|http://docs.mongodb.org/manual/core/aggregation-pipeline/>
+documents. Each pipeline document must be a hash reference. Not all
+pipeline stages are supported after C<$changeStream>.
+
+The optional second argument is a hash reference with options:
+
+=for :list
+* C<fullDocument> - The fullDocument to pass as an option to the
+  C<$changeStream> stage. Allowed values: C<default>, C<updateLookup>.
+  Defaults to C<default>.  When set to C<updateLookup>, the change
+  notification for partial updates will include both a delta describing the
+  changes to the document, as well as a copy of the entire document that
+  was changed from some time after the change occurred.
+* C<resumeAfter> - The logical starting point for this change stream.
```

Comment:
Please add to this that it must be the `_id` from a document returned from calling next on the ChangeStream object.